app/uk/gov/hmrc/transitmovementsrouter/models/MessageId.scala

```diff
@@ -16,4 +16,12 @@
 
 package uk.gov.hmrc.transitmovementsrouter.models
 
+import play.api.libs.json.Reads
+import play.api.libs.json.__
+
 case class MessageId(value: String) extends AnyVal
+
+object MessageId {
+
+  implicit val messageIdReads: Reads[MessageId] = (__ \ "messageId").read[String].map(MessageId(_))
```

Comment:
I suspect this'll get confusing if we try to either:

* Expand the response object
* Try to read the MessageId from elsewhere

I think I'd prefer to see a separate response object that contains the message ID, perhaps

```scala
case class PersistenceResponse(messageId: MessageId)
```

then we can just use

```scala
implicit val persistenceResponseReads = Json.reads[PersistenceResponse]
```

or whatever it is, return that from the connector, and get the message ID out where we need it.

Reply:
Good point Dan, made the update